src/Widgets/DisplaysOverlay.vala

```diff
@@ -384,116 +433,62 @@ public class Display.DisplaysOverlay : Gtk.Overlay {
-                case_2 = is_x_smaller_absolute (case_2, case_2_t) && !diagonally ? case_2 : case_2_t;
-                snap_x = true;
-                move = true;
-            } else if (is_projected (child_x, child_width, anchor_x, anchor_width)) {
-                debug ("Child is on the Y axis of Anchor %s\n", anchor.virtual_monitor.get_display_name ());
-                case_3 = is_x_smaller_absolute (case_3, case_3_t) && !diagonally ? case_3 : case_3_t;
-                case_4 = is_x_smaller_absolute (case_4, case_4_t) && !diagonally ? case_4 : case_4_t;
-                snap_y = true;
-                move = true;
-            } else {
-                debug ("Child is diagonally of Anchor %s\n", anchor.virtual_monitor.get_display_name ());
-                if (!move) {
-                    diagonally = true;
-                    case_1 = is_x_smaller_absolute (case_1, case_1_t) ? case_1 : case_1_t;
-                    case_2 = is_x_smaller_absolute (case_2, case_2_t) ? case_2 : case_2_t;
-                    case_3 = is_x_smaller_absolute (case_3, case_3_t) ? case_3 : case_3_t;
-                    case_4 = is_x_smaller_absolute (case_4, case_4_t) ? case_4 : case_4_t;
-                }
-            }
-        }
-
-        int shortest_x = is_x_smaller_absolute (case_1, case_2) ? case_1 : case_2;
-        int shortest_y = is_x_smaller_absolute (case_3, case_4) ? case_3 : case_4;
 
-        // X Snapping
-        if (!snap_y || is_x_smaller_absolute (shortest_x, shortest_y)) {
-            if (snap_x & move) {
-                debug ("moving child %d on X\n", shortest_x);
-                if (shortest_x < SNAP_LIMIT) ((DisplayWidget) child).set_geometry (child_x - shortest_x, child_y , child_width, child_height);
-            }
-        }
-
-        // Y Snapping
-        if (!snap_x || is_x_smaller_absolute (shortest_y, shortest_x)) {
-            if (snap_y & move) {
-                debug ("moving child %d on Y\n", shortest_y);
-                if (shortest_y < SNAP_LIMIT) ((DisplayWidget) child).set_geometry (child_x , child_y - shortest_y, child_width, child_height);
+            var distance_widget_anchor_x = anchor_x - widget_x;
+            var distance_widget_anchor_y = anchor_y - widget_y;
+            var distance_left   = distance_widget_anchor_x + anchor_width;
+            var distance_right  = distance_widget_anchor_x - widget_width;
+            var distance_top    = distance_widget_anchor_y + anchor_height;
+            var distance_bottom = distance_widget_anchor_y - widget_height;
+            var test_distance_x = distance_right > -distance_left ? distance_right : distance_left;
+            var test_distance_y = distance_bottom > -distance_top ? distance_bottom : distance_top;
+
+            if (distance_left > 0 && distance_right < 0) {
+                test_distance_x = 0;
+            } else if (distance_top > 0 && distance_bottom < 0) {
```

Comment:
Comment that this meams widget is between top and bottom edges of anchor so no vertical movement needed.